### DISCUSSION TITLE
Is it possible to add support for Refact LLM(1.6B)?

### DISCUSSION THREAD

**Discussion by Jipok** [ggml-org/llama.cpp#3013] [Q&A]
Reddit announce: https://www.reddit.com/r/LocalLLaMA/comments/169yonh/we_trained_a_new_16b_parameters_code_model_that/
Blog: https://refact.ai/blog/2023/introducing-refact-code-llm/
Code: https://github.com/smallcloudai/refact/
Model: https://huggingface.co/smallcloudai/Refact-1_6B-fim

Do I understand correctly that this model cannot yet be used in llama.cpp since there is no support for [Multi Query Attention](https://arxiv.org/abs/1911.02150) yet?
Is this the only blocker?

**Comment by Jipok** [comment]
The model is very desirable if you believe the benchmarks.
![изображение](https://github.com/ggerganov/llama.cpp/assets/25588359/4b446c4d-6769-467f-b32e-b681a9463d11)

**Comment by Green-Sky** [comment]
> Do I understand correctly that this model cannot yet be used in llama.cpp since there is no support for [Multi Query Attention](https://arxiv.org/abs/1911.02150) yet?
Is this the only blocker?

llama.cpp does support it. But the main code does not yet work with alibi.

**Comment by Jipok** [comment]
There are bounty($2000) for CPU inference support for Refact LLM: https://github.com/smallcloudai/refact/issues/77

**Comment by Green-Sky** [comment]
tracking issue: https://github.com/ggerganov/llama.cpp/issues/3061
